examples/shell.rs

```diff
@@ -0,0 +1,252 @@
+//! A simple shell demo for embedded-sdmmc
+//!
+//! Presents a basic command prompt which implements some basic MS-DOS style shell commands.
+
+use std::io::prelude::*;
+
+use embedded_sdmmc::{Directory, Error, Volume, VolumeIdx, VolumeManager};
+
+use crate::linux::{Clock, LinuxBlockDevice};
+
+mod linux;
+
+struct VolumeState {
+    directory: Directory,
+    volume: Volume,
+    path: Vec<String>,
+}
+
+struct Context {
+    volume_mgr: VolumeManager<LinuxBlockDevice, Clock, 8, 8, 4>,
+    volumes: [Option<VolumeState>; 4],
+    current_volume: usize,
+}
+
+impl Context {
+    fn current_path(&self) -> Vec<String> {
+        let Some(s) = &self.volumes[self.current_volume] else {
+            return vec![];
+        };
+        s.path.clone()
+    }
+
+    fn process_line(&mut self, line: &str) -> Result<(), Error<std::io::Error>> {
+        if line == "help" {
```

Comment:
FYI, https://docs.rs/logos/latest/logos/ is really nice for things like this :)

Reply:
Interesting! I'll check it out.